File: turi/src/main/java/com/turi/user/domain/port/UserService.java
package com.turi.user.domain.port;

import com.turi.user.domain.model.RefreshToken;
import com.turi.user.domain.model.ResetToken;
import com.turi.user.domain.model.User;

public interface UserService
{
    User getById(final Long id);

    User getByUsername(final String username);

    User getByEmail(final String email);

    String getUsername(final Long id);

    String getEmail(final Long id);

    User getByPasswordResetToken(final String resetToken);

    Long getUserIdByLogin(final String login);

    Boolean isUsernameExists(final String username);

    Boolean isEmailExists(final String email);

    ResetToken sendResetPasswordCode(final String email);

    RefreshToken resetPassword(final String resetToken, final Integer code);

    User create(final User user);

    void changeUsername(final Long id, final String username);

    void changeEmail(final Long id, final String email);

    void changePassword(final Long id, final String password);

    void deleteAllExpiredPasswordResetDetails();
}

File: turi-ui/src/services/userService.ts
import { API_BASE_URL } from '../config/api'
import { API } from './constants.ts'

export const userService = {
    getUsername: async () => {
        return await fetch(`${API_BASE_URL}${API.USER.GET_USERNAME}`, {
            method: 'GET',
            headers: {
                'Content-Type': 'application/json'
            },
            credentials: 'include'
        })
    },

    getEmail: async () => {
        return await fetch(`${API_BASE_URL}${API.USER.GET_EMAIL}`, {
            method: 'GET',
            headers: {
                'Content-Type': 'application/json'
            },
            credentials: 'include'
        })
    },

    checkIsUsernameExists: async (username: string) => {
        return await fetch(`${API_BASE_URL}${API.USER.IS_USERNAME_EXISTS}?username=${username}`, {
            method: 'GET',
            headers: {
                'Content-Type': 'application/json'
            },
            credentials: 'include'
        })
    },

    checkIsEmailExists: async (email: string) => {
        return await fetch(`${API_BASE_URL}${API.USER.IS_EMAIL_EXISTS}?email=${email}`, {
            method: 'GET',
            headers: {
                'Content-Type': 'application/json'
            },
            credentials: 'include'
        })
    },

    sendResetPasswordCode: async (email: string) => {
        return await fetch(`${API_BASE_URL}${API.USER.SEND_RESET_PASSWORD_CODE}?email=${email}`, {
            method: 'POST',
            headers: {
                'Content-Type': 'application/json'
            },
            credentials: 'include'
        })
    },

    resetPassword: async (code: string) => {
        return await fetch(`${API_BASE_URL}${API.USER.RESET_PASSWORD}?code=${code}`, {
            method: 'POST',
            headers: {
                'Content-Type': 'application/json'
            },
            credentials: 'include'
        })
    },

    changeUsername: async (username: string) => {
        return await fetch(`${API_BASE_URL}${API.USER.CHANGE_USERNAME}?username=${username}`, {
            method: 'PUT',
            headers: {
                'Content-Type': 'application/json'
            },
            credentials: 'include'
        })
    },

    changeEmail: async (email: string) => {
        return await fetch(`${API_BASE_URL}${API.USER.CHANGE_EMAIL}?email=${email}`, {
            method: 'PUT',
            headers: {
                'Content-Type': 'application/json'
            },
            credentials: 'include'
        })
    },

    changePassword: async (password: string) => {
        return await fetch(`${API_BASE_URL}${API.USER.CHANGE_PASSWORD}?password=${password}`, {
            method: 'PUT',
            headers: {
                'Content-Type': 'application/json'
            },
            credentials: 'include'
        })
    }
}
